schema.go

```diff
@@ -590,23 +368,24 @@ func serializeResource(rawResource any, info serializationInfo) (schema.Resource
 			continue
 		}
 
-		isInput := field.Type.Implements(reflect.TypeOf(new(pulumi.Input)).Elem())
-		_, isOutput := vField.Interface().(pulumi.Output)
+		isInputType := field.Type.Implements(reflect.TypeOf(new(pulumi.Input)).Elem())
+		_, isOutputType := vField.Interface().(pulumi.Output)
 
 		for fieldType.Kind() == reflect.Ptr {
 			fieldType = fieldType.Elem()
 		}
-		if isOutput {
+		//Need to fetch the underlying types for input and outputs
```

Comment:
Nit: Comments should have space after the `//`. This applies in multiple places.